0.35.1 release notes (#3917)
astronomer/docs: software/release-notes.md

:::

## 0.35.1

Release date: July 15, 2024

### Security fix for DAG-only deploys

> Fixed an issue where if you had a DAG-only deploy, it was possible for users to access its dag-server from any other kubernetes namespace. See [Upgrade considerations for 0.35](https://www.astronomer.io/docs/software/upgrade-astronomer#upgrade-to-astronomer-software-035) for more details about the security fix.

Author: ```suggestion
Fixed an issue where if you had DAG-only deploys enabled, it was possible for users to access its dag-server from any other kubernetes namespace. See [Upgrade considerations for 0.35](https://www.astronomer.io/docs/software/upgrade-astronomer#upgrade-to-astronomer-software-035) for more details about the security fix.
```

> Fixed an issue where if you had DAG-only deploys enabled, it was possible for users to access DAGs across different namespaces or Airflow deployments without proper permissions. See [Upgrade considerations for 0.34 and 0.35](https://www.astronomer.io/docs/software/upgrade-astronomer#upgrade-to-astronomer-software-034-or-035) for more details about the security fix.

Reviewer: ```suggestion
Fixed an issue where if you had a DAG-only deploy, it was possible for users to access its dag-server from any other kubernetes namespace. See [Upgrade considerations for 0.34 and 0.35](https://www.astronomer.io/docs/software/upgrade-astronomer#upgrade-to-astronomer-software-034-or-035) for more details about the security fix.
```

:::

## 0.35.1

Release date: July 15, 2024

### Security fix for DAG-only deploys

> Fixed an issue where if you had DAG-only deploys enabled, it was possible for users to access DAGs across different namespaces or Airflow deployments without proper permissions. Each pod requires a network policy to prevent cross-namespace interactions. This policy was not added to the DAG server component, leading to the security implication of cross-namespace DAG access. This issue occurs in Software versions 0.34.0-0.34.2 and 0.35. To mitigate the issue you can:

Reviewer 2: How much do we need to go down on the severity of the bug fix in public facing notes? Can mentioning the capability this bug enables potentially attract bad actors in itself as this is now in public notes
`it was possible for users to access DAGs across different namespaces or Airflow deployments without proper permissions`
cc: @danielhoherd @tonyHuinker

> Fixed an issue where if you had DAG-only deploys enabled, it was possible for users to access DAGs across different namespaces or Airflow deployments without proper permissions. Each pod requires a network policy to prevent cross-namespace interactions. This policy was not added to the DAG server component, leading to the security implication of cross-namespace DAG access. This issue occurs in Software versions 0.34.0-0.34.2 and 0.35. To mitigate the issue you can:

Author: ```suggestion
Fixed an issue where if you had DAG-only deploys enabled, it was possible for users to access DAGs across different namespaces or Airflow deployments without proper permissions. This issue occurs in Software versions 0.34.0-0.34.2 and 0.35. To mitigate the issue you can:
```
How does this look? A similar phrasing is also in the [upgrade considerations](https://github.com/astronomer/docs/pull/3917/files#diff-cc9f2660bf7ec0036ab0ace29568f4a1f3cc08f023e59da9909235a0ba1577acR225) and might need to be reworded as well.

Reviewer 2: This can be omitted - `This issue occurs in Software versions 0.34.0-0.34.2 and 0.35.0`

Author: Removed thanks!

Author: @danielhoherd - can you please review this most recent version? Looks like your comment is on an outdated one.

- Not enable DAG-only deploys in your Deployments
- Upgrade to 0.35.1 from 0.35
- If you are not using Software versions 0.34.0-0.34.2 or 0.35, wait to upgrade until the next major release.

> 

Reviewer 2: these can be omitted as well

:::

## 0.35.1

Release date: July 16, 2024

### Bug fixes

> - Fixed a bug where if you had DAG-only deploys enabled, it was possible for users in one Deployment to access DAGs in other Deployments, even if the user did not have access to other DAG-only Deployments.

Author: @danielhoherd - how does this wording look?

Reviewer: No, I don't think that wording accurately describes the problem.

> Fixed a bug where a DAG-only deploy dag-server was accessible from any kubernetes namespace.

    - [CVE-2022-21698](https://nvd.nist.gov/vuln/detail/CVE-2022-21698)

    - [CVE-2023-1370](https://www.cve.org/CVERecord?id=CVE-2023-1370)
    - [CVE-2023-39325](https://nvd.nist.gov/vuln/detail/CVE-2023-39325)
    - [CVE-2023-45283](https://nvd.nist.gov/vuln/detail/CVE-2023-45283)
    - [CVE-2023-45288](https://nvd.nist.gov/vuln/detail/CVE-2023-45288)
    - [CVE-2023-50868](https://nvd.nist.gov/vuln/detail/CVE-2023-50868)
    - [CVE-2023-50387](https://nvd.nist.gov/vuln/detail/CVE-2023-50387)

    - [CVE-2024-2961](https://nvd.nist.gov/vuln/detail/CVE-2024-2961)
    - [CVE-2024-21626](https://nvd.nist.gov/vuln/detail/CVE-2024-21626)
    - [CVE-2024-24790](https://nvd.nist.gov/vuln/detail/CVE-2024-24790)
    - [CVE-2024-25710](https://nvd.nist.gov/vuln/detail/CVE-2024-25710)
    - [CVE-2024-26308](https://nvd.nist.gov/vuln/detail/CVE-2024-26308)
    - [CVE-2024-28757](https://nvd.nist.gov/vuln/detail/CVE-2024-28757)
    - [CVE-2024-33599](https://nvd.nist.gov/vuln/detail/CVE-2024-33599)

    - [GHSA-xpw8-rcwv-8f8p](https://nvd.nist.gov/vuln/detail/CVE-2023-44487)
    - [GHSA-36jr-mh4h-2g58](https://security.snyk.io/vuln/SNYK-JS-D3COLOR-1076592)

>     - [GHSA-m425-mq94-257g]() -> https://nvd.nist.gov/vuln/detail/CVE-2023-44487 ?

Author: @himabindu07 -I wasn't sure if i should link to the underlying CVE or to the `go` issue: https://github.com/grpc/grpc-go/security/advisories/GHSA-m425-mq94-257g
